crates/topos-tce-transport/src/lib.rs

```diff
@@ -101,14 +114,80 @@ pub enum ProtocolEvents {
     /// Indicates that 'echo' message broadcasting is required
     Echo {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// Indicates that 'ready' message broadcasting is required
     Ready {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// For simulation purpose, for now only caused by ill-formed sampling
     Die,
 
     /// Stable Sample
     StableSample,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum Error {}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct ValidatorId(H160);
+
+impl ValidatorId {
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    pub fn address(&self) -> Address {
+        Address::from_str(&format!("0x{}", hex::encode(self.0))).unwrap()
+    }
+}
+
+impl From<H160> for ValidatorId {
+    fn from(address: H160) -> Self {
+        ValidatorId(address)
+    }
+}
+
+impl From<&str> for ValidatorId {
```

Comment:
Switch to `TryFrom` if possible to avoid `expect`

Reply:
In this case, we want to abort however. This function is used when parsing the validatos from `genesis.json` or from the `CLI`. So in case we can't parse the provided data, we should exit right away.